README.md

```diff
@@ -127,15 +127,34 @@ The function takes a single argument which is the `errorformat`.
 ### from_pattern
 
 ```lua
-parser = require('lint.parser').from_pattern(pattern, skeleton)
+parser = require('lint.parser').from_pattern(pattern, groups, severity_map, defaults)
 ```
 
-The function takes two arguments. The first is a lua pattern that must match
-four groups in order: line number, offset, code and message.
+The function allows to parse the linter's output using a lua regex pattern.
 
-The second argument is a skeleton used to create each diagnostic. This can be
-used to provide default values - for example to set a `severity`.
+- pattern: The regex pattern applied on each line of the output
+- groups: The groups specified by the pattern
 
+``` lua
+groups = {"lineno", "message", ("colno" | ("colbeg", "colend")), ["code"], ["codeDesc"], ["file"], ["severity"]}
```

Comment:
```suggestion
groups = {"lineno", "message",  "start_col", ["end_col"], ["code"], ["codeDesc"], ["file"], ["severity"]}
```

`start_col` and `end_col` is what's typically used within the neovim code base. I also wouldn't distinguish between `colno` and `start_col`, but just have the former, and default `end_col` to `start_col` if it isn't present.